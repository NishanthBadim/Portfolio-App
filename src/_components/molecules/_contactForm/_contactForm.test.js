import React from "react";
import { create } from "react-test-renderer";
import { render, fireEvent, getByTestId } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import ContactForm from "./_contactForm";

describe("Contact Form testing", () => {
  test("Matches the snapshot", () => {
    const input = create(<ContactForm />);
    expect(input.toJSON()).toMatchSnapshot();
  });

  test("Provide Valid Inputs to the Contact Form", () => {
    const { getByPlaceholderText, getByRole ,getByTestId} = render(<ContactForm />);
    fireEvent.change(getByPlaceholderText("Enter your name"), {
        target: { value: "Johnson" },
      });
      fireEvent.change(getByPlaceholderText("Enter your email address"), {
        target: { value: "john@gmail.com" },
      });
      fireEvent.change(getByPlaceholderText("Enter your phone number"), {
        target: { value: "1234567890" },
      });
      fireEvent.change(getByPlaceholderText("Enter the subject"), {
        target: { value: "hello" },
      });
    fireEvent.click(getByRole("button"));
    expect(getByTestId("emailError").value).toBeUndefined(); 
 });

  test("Provide Empty Inputs", () => {
    const { getByRole ,getByTestId} = render(<ContactForm />);

    fireEvent.click(getByRole("button"));
    expect(getByTestId("emailError")).toBeInTheDocument();
  });

  test("Provide Inavlid Inputs", () => {
    const { getByPlaceholderText, getByRole ,getByTestId} = render(<ContactForm />);
    fireEvent.change(getByPlaceholderText("Enter your name"), {
        target: { value: "12345" },
      });
      fireEvent.change(getByPlaceholderText("Enter your email address"), {
        target: { value: "xyz" },
      });
      fireEvent.change(getByPlaceholderText("Enter your phone number"), {
        target: { value: "atoz" },
      });
    fireEvent.click(getByRole("button"));
    expect(getByTestId("emailError")).toBeInTheDocument(); 
 });



});