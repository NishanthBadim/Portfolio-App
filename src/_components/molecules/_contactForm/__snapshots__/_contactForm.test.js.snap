// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Form testing Matches the snapshot 1`] = `
<div
  className="makeStyles-form-1"
>
  <form>
    <div
      className="makeStyles-field-2"
    >
      <div
        className="makeStyles-root-7"
      >
        <p
          className="MuiTypography-root MuiTypography-body2"
        >
          Name
        </p>
      </div>
      <div>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-input-8 MuiInputBase-fullWidth MuiInput-fullWidth"
          onClick={[Function]}
        >
          <input
            className="MuiInputBase-input MuiInput-input"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Enter your name"
            type="text"
          />
        </div>
      </div>
      <p
        className="MuiTypography-root makeStyles-error-6 MuiTypography-body1"
      />
    </div>
    <div
      className="makeStyles-field-2"
    >
      <div
        className="makeStyles-root-7"
      >
        <p
          className="MuiTypography-root MuiTypography-body2"
        >
          Email
        </p>
      </div>
      <div>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-input-8 MuiInputBase-fullWidth MuiInput-fullWidth"
          onClick={[Function]}
        >
          <input
            className="MuiInputBase-input MuiInput-input"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Enter your email address"
            type="email"
          />
        </div>
      </div>
      <p
        className="MuiTypography-root makeStyles-error-6 MuiTypography-body1"
        data-testid="emailError"
      />
    </div>
    <div
      className="makeStyles-field-2"
    >
      <div
        className="makeStyles-root-7"
      >
        <p
          className="MuiTypography-root MuiTypography-body2"
        >
          Phone number
        </p>
      </div>
      <div>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-input-8 MuiInputBase-fullWidth MuiInput-fullWidth"
          onClick={[Function]}
        >
          <input
            className="MuiInputBase-input MuiInput-input"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Enter your phone number"
            type="text"
          />
        </div>
      </div>
      <p
        className="MuiTypography-root makeStyles-error-6 MuiTypography-body1"
      />
    </div>
    <div
      className="makeStyles-field-2"
    >
      <div
        className="makeStyles-root-7"
      >
        <p
          className="MuiTypography-root MuiTypography-body2"
        >
          Subject
        </p>
      </div>
      <div>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-input-8 MuiInputBase-fullWidth MuiInput-fullWidth"
          onClick={[Function]}
        >
          <input
            className="MuiInputBase-input MuiInput-input"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Enter the subject"
            type="text"
          />
        </div>
      </div>
      <p
        className="MuiTypography-root makeStyles-error-6 MuiTypography-body1"
      />
    </div>
    <div
      className="makeStyles-buttonBox-4"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-5"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Submit
        </span>
      </button>
    </div>
  </form>
</div>
`;
